### PR TITLE
Add Hive session property for max partitions per writer

### PR DESCRIPTION
**Pull request by zhenxiao**

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java [hunk 1]
```diff
@@ -442,6 +448,11 @@ public static boolean isCollectColumnStatisticsOnWrite(ConnectorSession session)
         return session.getProperty(COLLECT_COLUMN_STATISTICS_ON_WRITE, Boolean.class);
     }
 
+    public static int getMaxPartitionsPerWriter(ConnectorSession session)
```
Comment (sopel39) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java]: usage of this method is missing. Have you forgot to push it?

Comment (zhenxiao) [reply presto-hive/src/main/java/com/facebook/presto/hive/HiveSessionProperties.java]: oh, yes. Get it added now. thank you @sopel39

File: presto-hive/src/main/java/com/facebook/presto/hive/HivePageSinkProvider.java [hunk 2]
```diff
@@ -159,7 +157,7 @@ private ConnectorPageSink createPageSink(HiveWritableTableHandle handle, boolean
                 pageIndexerFactory,
                 typeManager,
                 hdfsEnvironment,
-                maxOpenPartitions,
+                hiveSessionProperties.getMaxPartitionsPerWriter(session),
```
Comment (findepi) [inline presto-hive/src/main/java/com/facebook/presto/hive/HivePageSinkProvider.java]: this is a static method

**Comment by zhenxiao** [comment]
thank you @findepi 
get it fixed

**Comment by zhenxiao** [comment]
@wenleix we are setting up one dedicated Presto ETL cluster to support a few expensive data ingestion. Always hitting max writer limit. Adding this property could remove repeated restart of our cluster. What do you think?

**Comment by dain** [comment]
@zhenxiao, if you have a dedicated cluster, then why not set the value to 1000000, and effectively disable it?

**Comment by zhenxiao** [comment]
@dain OK. was thinking adding this property to provide some flexibility

**Comment by dain** [comment]
@findepi, @sopel39: how do you feel about adding this, with regards to the comments from @wenleix

**Comment by sopel39** [comment]
@electrum Wouldn't the same argument be applicable to `hive.max-partitions-per-scan` property (saturating Metastore that affects whole system). I was thinking of something similar (global cap), but then only privileged users are allowed to bump this property.

**Pull request closed without merging**